### PR TITLE
fix(cache) support external redis which need password

### LINKED ISSUE #90: Support external redis which need password
Now if the external redis don't need password, the pod of jobservice can be running.
If the external redis need password, the pod will be crash.

### PR DESCRIPTION
**Pull request by Ghostbaby**
Support external redis which need password

Fixes #90

Signed-off-by: zhuhuijun <zhuhuijunzhj@gmail.com>

### CODE REVIEW

File: controllers/cache/readiness.go [hunk 1]
```diff
@@ -138,15 +137,16 @@ func (redis *RedisReconciler) GetExternalRedisInfo() (*rediscli.Client, error) {
 		}
 
 		connect = &RedisConnect{
-			Endpoint:  strings.Join(endpoint[:], ","),
+			Endpoint:  endpoint,
 			Port:      port,
 			Password:  pw,
 			GroupName: spec.GroupName,
+			Schema:    RedisSentinelSchema,
```
Comment (wangcanfengxs) [inline controllers/cache/readiness.go:144]: the schema is `sentinel`?

File: controllers/cache/utils.go [hunk 2]
```diff
@@ -257,7 +257,36 @@ func (redis *RedisReconciler) GetPodsStatus(podArray []corev1.Pod) ([]corev1.Pod
...
+		return ""
+	}
+}
+
+// genRedisSentinelConnURL returns redis sentinel connection url
+func (c *RedisConnect) genRedisSentinelConnURL() string {
+
+	hostInfo := GenHostInfo(c.Endpoint, c.Port)
+	if c.Password != "" {
+		return fmt.Sprintf("redis+sentinel://:%s@%s/mymaster/0", c.Password, hostInfo)
```
Comment (wangcanfengxs) [inline controllers/cache/utils.go:275]: Where does the `Schema` use?

Comment (Ghostbaby) [reply controllers/cache/utils.go:275]: `GenRedisConnURL` will check schema type.

File: controllers/cache/readiness.go (hunk 1 above)
Comment (Ghostbaby) [reply controllers/cache/readiness.go:144]: yes

**Pull request merged**